cloud.foundry.cli/src/test/java/cloud/foundry/cli/system/GetAllCommandTest.java

```diff
@@ -0,0 +1,142 @@
+
+        SpecBean spec = rootBean.getSpec();
+        // TODO: check if we can put in an absolute value once we use "throw-away spaces" created per test run
+        assert spec.getSpaceDevelopers().size() > 0;
+    }
+
+    private ConfigBean loadConfigBeanFromStdoutAndAssertItsValidity(String stdoutContent, SpaceManager spaceManager) {
+        ConfigBean rootBean = YamlMapper.loadBeanFromString(stdoutContent, ConfigBean.class);
+        assertRootBeanIsValid(rootBean, spaceManager);
+        return rootBean;
+    }
+
+    @Test
+    public void testEmptySpace() {
+        try (SpaceManager spaceManager = createSpaceWithRandomName()) {
+            ArgumentsBuilder args = new ArgumentsBuilder()
+                    .addArgument("get");
+
+            RunResult runResult = runBaseControllerWithCredentialsFromEnvironment(args, spaceManager);
+
+            String outContent = runResult.getStreamContents().getStdoutContent();
+            assert outContent.length() > 0;
+
+            // parse YAML from stdout and check it for validity
+            ConfigBean rootBean = loadConfigBeanFromStdoutAndAssertItsValidity(outContent, spaceManager);
+
+            SpecBean spec = rootBean.getSpec();
+            assert spec.getApps() == null;
+            assert spec.getServices() == null;
+            assert spec.getSpaceDevelopers().size() > 0;
+
+            // TODO: check log contents
+            String errContent = runResult.getStreamContents().getStderrContent();
+            assert errContent.length() > 0;
+
+            assert runResult.getExitCode() == 0;
+        }
+    }
+
+    @Test
+    public void testSpaceWithOneService() {
+        try (SpaceManager spaceManager = createSpaceWithRandomName()) {
+            ServiceBean service = new ServiceBean();
+            service.setService("elephantsql");
+            service.setPlan("turtle");
+            String serviceName = spaceManager.requestCreationOfService(service);
+            spaceManager.createRequestedEntities();
+
+            ArgumentsBuilder args = new ArgumentsBuilder()
+                    .addArgument("get");
+
+            RunResult runResult = runBaseControllerWithCredentialsFromEnvironment(args, spaceManager);
+
+            String outContent = runResult.getStreamContents().getStdoutContent();
+            assert outContent.length() > 0;
+
+            // parse YAML from stdout and check it for validity
+            ConfigBean rootBean = loadConfigBeanFromStdoutAndAssertItsValidity(outContent, spaceManager);
+
+            SpecBean spec = rootBean.getSpec();
+            assert spec.getApps() == null;
+
+            assert spec.getServices().size() == 1;
+            ServiceBean parsedService = spec.getServices().get(serviceName);
+            assert parsedService.getService().equals(service.getService());
+            assert parsedService.getPlan().equals(service.getPlan());
+            assert parsedService.getTags() == null;
+
+            // TODO: check log contents
```

Comment:
Again, is it planned to address this TODO? It not we should remove it.

Reply:
We could perform a veeery basic test there, the exact logging is handled in regular unit tests.

Reply:
There's no log messages >= INFO level issued during our test, only some ref-related ones on DEBUG and the setup/teardown stuff. I'll replace them with a comment explaining why we don't check the messages in-depth.